src/Pos/Block/Logic/Internal.hs

```diff
@@ -111,8 +114,12 @@ type BlockApplyMode ssc m
 applyBlocksUnsafe
     :: forall ssc m . BlockApplyMode ssc m
     => OldestFirst NE (Blund ssc) -> Maybe PollModifier -> m ()
-applyBlocksUnsafe blunds0 pModifier =
-    reportingFatal version $
+applyBlocksUnsafe blunds0 pModifier = reportingFatal version $ do
+    when (length blunds /= length blunds0) $ assertionFailed $
```

Comment:
I can instead check that `b` is null in  `(_,b) = spanSafe ...`. Will it be better?